### ISSUE TITLE
Realtime notification to join team

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/action#1151]
## Enhancement

When a user is actively using a Parabol view in browser and gets invited to join a team, they want to see a notification immediately so that they can take care of the task without changing tools/tabs.

Currently they get an email with a button to join the new team.

- We could create a user notification where they can tap to join.
- We could pop a toast where they could tap to join.
- We could add this to notification settings where users can throttle when and why they get email, notifications, etc. This would probably be part of an epic around notifications and email.

**Estimated effort**: 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by jordanh** [comment]
I'd like to elevate the priority of this one and schedule it for the next Epic. I was working with a team this morning and our invitation emails kept being eaten by their spam server.

**Comment by ackernaut** [comment]
👍

**Comment by mattkrick** [comment]
This is not session-based so I'm gonna rule out toasts. So let's use notifications.
From the team settings, a user ("inviter") invites an email address ("invitee").
- The inviter could be an Org billing leader, or not
- The invitee could be on parabol, or not
- If in parabol, the invitee could be in the org, or not
- If in the org, the invitee could have previously been on the team, or not.

Combinatorics time!
https://docs.google.com/a/parabol.co/spreadsheets/d/1AHNUWc1dHovEVa0N3eok48_3C7yS8jW8rnd9ikk6WHI/edit?usp=sharing

**Comment by mattkrick** [comment]
From slack
```
1) I think it should resend the notification – I mean, if an invitee accidentally deletes the invitation notification it should be able to be resent, right?


[11:29] 
2) I think I like this logic – is there some tradeoff you are exploring?


[11:29] 
3) Oh, this is interesting. I guess I assumed I could delete the notification in #1 above. Yeah, I think there is an “Ignore” to delete the notification



----- Today September 13th, 2017 -----
terry
[7:00 AM] 
:eyes:


terry
[7:12 AM] 
1) So according to the matrix we worked on as a team, most folks are just getting a notification instead of an email, right? So if a person were to nag (we could consider removing any ability to nag), we could 1) check to see if the invitee cleared their notification without joining and send another notification, otherwise do nothing 2) if the inviter _really_ wanted to nag and the invitee did not receive the invite email (according to the matrix) we could send the invite email (and optionally remove the option to nag again—perhaps folks only get 1 nag per invite?) or 3) we could design a simple nag email (and I think give folks 1 opportunity to nag)
2) So basically the tension is around the ex-member having the opportunity to give consent? I think I like the idea of them having the choice to join opposed to it being automatic.
3) I like the idea of some “Deny” or “Ignore” button. (edited)


mattkrick [9:53 AM] 
@jrhusney 1) say you're the inviter, i'm the invitee. if i click ignore, then in team settings you'll see I'm still "invited" but now i have no way to join, so we should remove my name from that list of invitees. if we remove my name, you'll want to know about it, so we'll need to tell you "matt has denied your invite", so it seems like with this path, "deny" seems like a more transparent option. If we want to mimic the email flow, we'd just have a notification that can't be removed, which means i have to tell you in real life to cancel my invitation. (edited)


[9:54] 
an ignore flow would still work if we send both, because then if i ignore, i could still join the team via email.


[9:55] 
2) The alternative would be that I get the choice to rejoin the team instead of instantly added to it.


[9:59] 
@terry 1) i like nags, but i need to keep this _very_ bare bones since i'm out of time for this sprint. I think the lost cost sensible choices here are: 
1) Send both & add a deny button. 
2) send both & add an ignore button
3) Send just notification to actives & add a deny button
```

I'm gonna go with option 1. My goal is to make this simpler for the inviter at the cost of possibly annoying the invitee. Right now, inviters already can see 3 states in their Team Settings: People on the team, People invited, People awaiting approval. I don't want to add a 4th, because they shouldn't have to care whether the invite goes out via notification or email.
The tension is around the "Resend Invitation" button. If I ask Jordan, "hey, can you resend that invitation?" He should be able to click that button and say "OK sent! check your inbox". **How the invite arrives should be deterministic.** If it isn't, it makes us look bad (huh that's weird, i clicked invite & it didn't send an email). If i didn't log in for 29 days before you send me the invite, then i go inactive tomorrow, when you click Resend, you'll be sending the email for the first time. This is not intuitive & we shouldn't force the inviter to know our internal logic.

**Comment by jordanh** [comment]
👍👍👍

**Comment by mattkrick** [comment]
closed with epic 6.5
